src/main/java/com/commercetools/sync/services/impl/CustomObjectServiceImpl.java

```diff
@@ -131,8 +133,27 @@ public CustomObjectServiceImpl(@Nonnull final CustomObjectSyncOptions syncOption
         if (StringUtils.isEmpty(customObjectDraft.getKey()) || StringUtils.isEmpty(customObjectDraft.getContainer())) {
             return CompletableFuture.completedFuture(Optional.empty());
         }
-        return createResource(customObjectDraft,
-            draft -> CustomObjectCompositeIdentifier.of(draft).toString(),
-            CustomObjectUpsertCommand::of);
+        CompletionStage<Optional<CustomObject<JsonNode>>> createdResource = createResource(customObjectDraft,
+            draft -> CustomObjectCompositeIdentifier.of(draft).toString(), CustomObjectUpsertCommand::of);
+        return createdResource ;
+    }
+
+    @Nonnull
+    @Override
+    CompletionStage<Optional<CustomObject<JsonNode>>> executeCreateCommand(
```

Comment:
Why did you make this change? Why this method doesn't have `handle` anymore like the parent method?

Reply:
Because `CustomObject` is a bit different from other resources when doing update resources.
In sync class, no matter it attempts to either update CO method or create CO method, in the end it runs `BaseService.createResource()`. Any exception occurs in this method is handled and isn't thrown to Sync class. As a result, retry mechanism cannot proceed when updating CO. For this reason, `CustomObjectServiceImpl.executeCreateCommand()` is used to override and thrown exception can be managed in Sync class instead.